base/compiler/ssair/ir.jl

```diff
@@ -1145,7 +1145,7 @@ function process_node!(compact::IncrementalCompact, result_idx::Int, inst::Instr
         result_idx += 1
     elseif isa(stmt, GlobalRef)
         total_flags = IR_FLAG_CONSISTENT | IR_FLAG_EFFECT_FREE
-        if (result[result_idx][:flag] & total_flags) == total_flags
+        if (result[result_idx][:flag] & total_flags) ≠ 0
```

Comment:
I don't think this is right. I want both flags.

Reply:
Right. Now I changed the `else` branch so that we propagate the original `flag` to newly generated statement, but didn't change the behavior of the `then` branch.